### PR TITLE
Feat/star

### PR DESCRIPTION
**Pull request by ivyhu630**
Added factor bar below rating overview;
Fixed ratingOverview edge case when there's no rating or when there's no review for helpfulness: see product 65638 and 65635

### CODE REVIEW

File: client/src/components/Ratings/RatingBreakdown/FactorBar.jsx [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+import styled from 'styled-components';
+import PropTypes from 'prop-types';
+import FactorType from './FactorType';
+
+export default function FactorBar({ factor, score }) {
+  // caculate the left space based on lenth of factorbar(15rem)
+  const distanceToLeft = Math.floor((15 * score) / 5).toString().concat('rem');
+
+  return (
+    <FactorBarBlock>
```
Comment (floridamaniac) [inline client/src/components/Ratings/RatingBreakdown/FactorBar.jsx:11]: So semantic, love it

File: client/src/index.jsx [hunk 2]
```diff
@@ -33,7 +34,9 @@ function App() {
           <RelatedProducts />
         </StylesProvider>
         <QA />
-        <Ratings />
+        <RatingProvider>
```
Comment (floridamaniac) [inline client/src/index.jsx:37]: Can we move the RatingProvider to also include Overview & Related Products? We'll both need access to this context to get the average rating for the star component.

File: client/src/components/Ratings/Ratings.jsx [hunk 3]
```diff
@@ -1,21 +1,22 @@
...
-import { RatingProvider } from '../../contexts/ReviewMeta';
+import { useMeta } from '../../contexts/ReviewMeta';
 
 import RatingBreakdown from './RatingBreakdown';
 import ReviewList from './ReviewList';
 import { ReviewProvider } from '../../contexts/ProductReview';
 
 export default function Ratings() {
+  const currentMeta = useMeta();
+  const ratingBreakdown = (currentMeta && currentMeta.avgRating) ? <RatingBreakdown /> : <div />;
```
Comment (floridamaniac) [inline client/src/components/Ratings/Ratings.jsx:11]: Interesting, is this a placeholder div in case it's loading/not available? I might steal this.

**Review by floridamaniac — APPROVED** [review]
The factor bars look IMMACULATE! Great PR.

File: client/src/components/Ratings/Ratings.jsx (hunk 3 above)
Comment (ivyhu630) [reply client/src/components/Ratings/Ratings.jsx:11]: Yep. It's a lazy way of double conditional render - This handles the edge case of when a product has no review and also bypass the async loading error of getting undefined

File: client/src/index.jsx (hunk 2 above)
Comment (ivyhu630) [reply client/src/index.jsx:37]: I just moved it in my local feat branch and it is giving me below warning even though code works. I'll take another look later in the day.
Screen Shot 2022-04-07 at 9 42 57 AM
.

**Pull request merged**